tools/envoy_headersplit/replace_includes_test.py

```diff
@@ -0,0 +1,73 @@
+# Lint as: python3
+"""Tests for replace_includes."""
+
+import unittest
+from unittest import mock
+import os
+from pathlib import Path
+import replace_includes
+
+
+class ReplaceIncludesTest(unittest.TestCase):
+
+  def test_to_classname(self):
+    # Test file name with whole path
+    self.assertEqual(replace_includes.to_classname("test/mocks/server/admin_stream.h"),
+                     "MockAdminStream")
+
+    # Test file name without .h extension
+    self.assertEqual(replace_includes.to_classname("cluster_mock_priority_set"),
+                     "MockClusterMockPrioritySet")
+
+  def test_to_bazelname(self):
+    # Test file name with whole path
+    self.assertEqual(replace_includes.to_bazelname("test/mocks/server/admin_stream.h", "server"),
+                     "//test/mocks/server:admin_stream_mocks")
+
+    # Test file name without .h extension
+    self.assertEqual(replace_includes.to_bazelname("cluster_mock_priority_set", "upstream"),
+                     "//test/mocks/upstream:cluster_mock_priority_set_mocks")
+
+  class FakeDir():
+    # fake directory to test get_filenames
+    def glob(sefl, _):
+      return [
+          Path("test/mocks/server/admin_stream.h"),
+          Path("test/mocks/server/admin.h"),
+          Path("test/mocks/upstream/cluster_manager.h")
+      ]
+
+  @mock.patch('replace_includes.Path', return_value=FakeDir())
+  def test_get_filenames(self, mock_Path):
+    self.assertEqual(replace_includes.get_filenames('sever'), [
+        'test/mocks/server/admin_stream.h', 'test/mocks/server/admin.h',
+        'test/mocks/upstream/cluster_manager.h'
+    ])
+
+  def test_replace_includes(self):
+    fake_source_code = open('code_corpus/fake_source_code.cc', 'r').read()
+    fake_build_file = open('code_corpus/fake_build', 'r').read()
+    os.mkdir('test')
+    os.mkdir('test/mocks')
+    os.mkdir('test/mocks/upstream')
```

Comment:
Are these tests run under Bazel? Is the tool itself run under Bazel? If not, how is this tested in CI?

Reply:
it's not run under Bazel now, I am working on moving them to be tested by Bazel now

Reply:
OK, great. LMK when that's ready and the other feedback has been applied, thanks.
/wait